syntax = "proto3";

package intel;
option csharp_namespace = "Intel.Proto";

service IntelWorker {
  rpc GetNewsCount (GetNewsRequest) returns (GetNewsReply);
  rpc GetSslGrade  (SslRequest)     returns (SslReply);
  rpc GetTechCount (TechRequest)    returns (TechReply);
  rpc GetTechBreakdown (TechRequest)        returns (TechBreakdownReply);
}

message GetNewsRequest { string company_name = 1; string domain = 2; }
message GetNewsReply   { int32 total = 1; repeated string sample_titles = 2; }

message SslRequest { string domain = 1; }
message SslReply   { string grade  = 1; }

message TechRequest { string domain = 1; }
message TechReply   { int32  count  = 1; }

message TechnologyItem {
  string category   = 1;
  string name       = 2;
  string first_seen = 3; // ISO-8601 string
  string last_seen  = 4; // ISO-8601 string
}

message TechBreakdownReply {
  repeated TechnologyItem items = 1;
}